components/collapsible-panel/collapsible-panel.js

```diff
@@ -0,0 +1,405 @@
+				content: '';
+				display: flex;
+				margin: 0.3rem;
+			}
+			.d2l-collapsible-panel-opener {
+				margin-inline-end: var(--d2l-collapsible-panel-spacing-inline);
+			}
+			.d2l-collapsible-panel-opener > d2l-icon-custom {
+				height: 0.9rem;
+				margin: 0.6rem;
+				margin-inline-end: 0;
+				position: relative;
+				width: 0.9rem;
+			}
+			.d2l-collapsible-panel-opener > d2l-icon-custom svg {
+				position: absolute;
+				transform-origin: 0.4rem;
+			}
+			:host([expanded]) .d2l-collapsible-panel-opener > d2l-icon-custom svg {
+				fill: var(--d2l-color-tungsten);
+				transform: rotate(90deg);
+			}
+			@media (prefers-reduced-motion: no-preference) {
+				.d2l-collapsible-panel-divider {
+					transition: opacity var(--d2l-collapsible-panel-transition-time) ease-in-out;
+				}
+				.d2l-collapsible-panel-opener > d2l-icon-custom svg {
+					animation: d2l-collapsible-panel-opener-close var(--d2l-collapsible-panel-arrow-time) ease-in-out;
+				}
+				:host([expanded]) .d2l-collapsible-panel-opener > d2l-icon-custom svg {
+					animation: d2l-collapsible-panel-opener-open var(--d2l-collapsible-panel-arrow-time) ease-in-out;
+				}
+				/* stylelint-disable order/properties-alphabetical-order */
+				@keyframes d2l-collapsible-panel-opener-open {
+					0% { transform: rotate(0deg); }
+					25% { transform: rotate(105deg); animation-timing-function: ease-in-out; }
+					50% { transform: rotate(82deg); animation-timing-function: ease-in-out; }
+					75% { transform: rotate(93deg); animation-timing-function: ease-in-out; }
+					100% { transform: rotate(90deg); animation-timing-function: ease-in-out; }
+				}
+				@keyframes d2l-collapsible-panel-opener-close {
+					0% { transform: rotate(90deg); }
+					25% { transform: rotate(-15deg); animation-timing-function: ease-in-out; }
+					50% { transform: rotate(8deg); animation-timing-function: ease-in-out; }
+					75% { transform: rotate(-3deg); animation-timing-function: ease-in-out; }
+					100% { transform: rotate(0deg); animation-timing-function: ease-in-out; }
+				}
+				/* stylelint-enable */
+			}
+			.d2l-collapsible-panel-divider {
+				border-bottom: 1px solid var(--d2l-color-mica);
+				margin-inline: var(--d2l-collapsible-panel-spacing-inline);
+				opacity: 1;
+			}
+			:host([type=inline]) .d2l-collapsible-panel-divider {
+				opacity: 0;
+			}
+			:host(:not([expanded])) .d2l-collapsible-panel:not(.has-summary) .d2l-collapsible-panel-divider {
+				opacity: 0;
+			}
+			.d2l-collapsible-panel-summary,
+			.d2l-collapsible-panel-content {
+				padding: 0.9rem var(--d2l-collapsible-panel-spacing-inline);
+			}
+			:host([type=inline]) .d2l-collapsible-panel-summary,
+			:host([type=inline]) .d2l-collapsible-panel-content {
+				padding-top: 0;
+			}
+			:host([type="inline"]) .d2l-collapsible-panel-summary {
+				margin-top: -0.3rem;
+			}
+			.d2l-collapsible-panel:not(.has-summary) .d2l-collapsible-panel-summary {
+				display: none;
```

Comment:
Alternate way to do this would be to put the `hidden` attribute on it when you render it.

Reply:
+1.  I like to avoid `:not(...)` - whenever I see it I usually look for a better way, since the `:not()` is effectively an extra selector check the browser needs to do, optimization that it might be doing aside.